src/foam/core/FObject.java

```diff
@@ -299,16 +299,16 @@ default FObject copyFrom(FObject obj) {
     List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
     for ( PropertyInfo p : props ) {
       try {
-        if ( p.isSet(obj) ) p.set(this, p.get(obj));
-      } catch (ClassCastException e) {
-        try {
+        if ( getClass().equals(obj.getClass()) ) {
```

Comment:
Use == instead. @penzital This is caused by the same instance variable overriding issue that you detected. If we merge this, we should create a ticket to fix the underlying issue and then revert this change.

Reply:
Fixed in https://github.com/kgrgreer/foam3/pull/1264/commits/a8836971de3d56bf006bac4900c8729b9a1cf40e.